service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/C21OrderEmailContentProviderTest.java

```diff
@@ -0,0 +1,139 @@
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.JURISDICTION;
+import static uk.gov.hmcts.reform.fpl.enums.JudgeOrMagistrateTitle.HER_HONOUR_JUDGE;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createC21Orders;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createDocumentReference;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createHearingBookings;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createJudgeAndLegalAdvisor;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createRespondents;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {JacksonAutoConfiguration.class, C21OrderEmailContentProvider.class,
+    HearingBookingService.class, LocalAuthorityNameLookupConfiguration.class, DateFormatterService.class})
+class C21OrderEmailContentProviderTest {
+    private final LocalDate date = LocalDate.now();
+    private final DateFormatterService dateFormatterService = new DateFormatterService();
+    private final HearingBookingService hearingBookingService = new HearingBookingService();
+
+    private static final String LOCAL_AUTHORITY_CODE = "example";
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @MockBean
+    private LocalAuthorityNameLookupConfiguration localAuthorityNameLookupConfiguration;
+
+    private C21OrderEmailContentProvider c21OrderEmailContentProvider;
+
+    private String familyManCaseNumber;
+
+    @BeforeEach
+    void setup() {
+        this.c21OrderEmailContentProvider = new C21OrderEmailContentProvider("",
+            objectMapper, hearingBookingService, localAuthorityNameLookupConfiguration, dateFormatterService);
+
+        given(localAuthorityNameLookupConfiguration.getLocalAuthorityName(LOCAL_AUTHORITY_CODE))
+            .willReturn("Example Local Authority");
+
+        familyManCaseNumber = RandomStringUtils.randomAlphabetic(8);
+    }
+
+    @Test
+    void shouldReturnExactC21NotificationParametersWithUploadedDocumentUrl() {
+        CaseDetails caseDetails = createCaseDetailsWithSingleC21Element();
+
+        final String subjectLine = "Jones, " + familyManCaseNumber;
+
+        Map<String, Object> returnedParameters = c21OrderEmailContentProvider.buildC21OrderNotification(caseDetails,
+            LOCAL_AUTHORITY_CODE);
+
+        assertThat(returnedParameters)
+            .extracting("subjectLine", "localAuthorityOrCafcass", "hearingDetailsCallout", "reference",
+                 "caseUrl")
+            .containsExactly(subjectLine, "Example Local Authority",
+                subjectLine + ", hearing " + dateFormatterService.formatLocalDateToString(date, FormatStyle.MEDIUM),
+                "167888", "/case/" + JURISDICTION + "/" + CASE_TYPE + "/167888");
+
+        /*  extracting separately as document_binary_url in populateCaseDetailsWithSingleC21Element()
+            method uses UUID.randomUUID()
+         */
```

Comment:
Don't use randomUUID? Or pass externally the UUID value which should be used? If our test is checking not equal to empty on unit test level it's always smell to me. I change the implementation to always generate ABC and is it going to be correct implementation? Unit tests should spot that.